### PR TITLE
Update data source `azurerm_management_group` - support get data source of mgmt group by using `display_name`

### LINKED ISSUE #6757: azurerm_management_group validation should allow spaces
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform version: 0.12.8
AzureRM version: 2.0.0

### Affected Resource(s)

* `azurerm_management_group`
* `data.azurerm_management_group`

### Debug Output

`can only consist of ASCII letters, digits, -, _, (, ), . , and cannot exceed the maximum length of 90`

Output as shown here: https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/managementgroup/validate/management_group.go#L17-L20

### Expected Behavior

Successful terraform plan

### Actual Behavior

Validation fails due to regex not allowing for a space in management group names

### Steps to Reproduce

1. Config:
```
data azurerm_management_group "foo" {
    name = "group with spaces"
}
```

2. `terraform plan`

### References

While the code does contain a comment about validation being shown in the portal, I couldn't find it. Management groups can still be created with spaces. Existing documentation suggests this is also the case too. E.g.: https://docs.microsoft.com/en-us/azure/governance/management-groups/create#create-in-portal

### PR DESCRIPTION
**Pull request by ArcturusZhang**
Fixes #6757

### CODE REVIEW

File: website/docs/r/management_group.html.markdown [hunk 1]
```diff
@@ -43,8 +43,12 @@ The following arguments are supported:
 
 * `group_id` - (Optional) The name or UUID for this Management Group, which needs to be unique across your tenant. A new UUID will be generated if not provided. Changing this forces a new resource to be created.
 
+~> **Note** `name` and `group_id` here corresponds to `ID` on Azure Portal.
+
 * `display_name` - (Optional) A friendly name for this Management Group. If not specified, this'll be the same as the `name`.
 
+~> **Note** `display_name` here corresponds to `Name` on Azure Portal.
```
Comment (tombuildsstuff) [inline website/docs/r/management_group.html.markdown:50]: The Azure Portal changes all the time - this isn't something we should be relying on here

```suggestion
```

File: website/docs/r/management_group.html.markdown [hunk 2]
```diff
@@ -43,8 +43,12 @@ The following arguments are supported:
 
 * `group_id` - (Optional) The name or UUID for this Management Group, which needs to be unique across your tenant. A new UUID will be generated if not provided. Changing this forces a new resource to be created.
 
+~> **Note** `name` and `group_id` here corresponds to `ID` on Azure Portal.
```
Comment (tombuildsstuff) [inline website/docs/r/management_group.html.markdown:46]: The Azure Portal changes all the time - this isn't something we should be relying on here



```suggestion
```

File: website/docs/d/management_group.html.markdown [hunk 3]
```diff
@@ -32,17 +36,19 @@ The following arguments are supported:
 
 ~> **NOTE:** The field `group_id` has been deprecated in favour of `name`.
 
+* `display_name` - Specifies the display name of this Management Group.
+
+~> **NOTE** As `display_name` is not unique errors may occur when there are multiple Management Groups with same display name.
```
Comment (tombuildsstuff) [inline website/docs/d/management_group.html.markdown:41]:
```suggestion
~> **NOTE** Whilst multiple management groups may share the same display name, when filtering Terraform expects a single management group to be found with this name. 
```

File: azurerm/internal/services/managementgroup/tests/management_group_data_source_test.go [hunk 4]
```diff
@@ -26,7 +26,25 @@ func TestAccDataSourceArmManagementGroup_basic(t *testing.T) {
...
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { acceptance.PreCheck(t) },
+		Providers: acceptance.SupportedProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceArmManagementGroup_basicByDisplayName(data),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(data.ResourceName, "display_name", fmt.Sprintf("acctestmg-%d", data.RandomInteger)),
+					resource.TestCheckResourceAttr(data.ResourceName, "subscription_ids.#", "0"),
```
[the remark this answers is not in the document]
Comment (ArcturusZhang) [reply azurerm/internal/services/managementgroup/tests/management_group_data_source_test.go:40]: I suppose for a data source we cannot add an import step, which will lead to test errors.

File: website/docs/d/management_group.html.markdown [hunk 5]
```diff
@@ -13,6 +13,10 @@ Use this data source to access information about an existing Management Group.
 ## Example Usage
 
 ```hcl
+provider "azurerm" {
+  features {}
+}
```
[the remark this answers is not in the document]
Comment (ArcturusZhang) [reply website/docs/d/management_group.html.markdown:18]: Sure, I will remove this. Is this also valid for the docs of resources? (A lot of them have already added this...)

File: azurerm/internal/services/managementgroup/management_group_data_source.go [hunk 6]
```diff
@@ -112,6 +125,37 @@ func dataSourceArmManagementGroupRead(d *schema.ResourceData, meta interface{})
...
+		}
+
+		if err := iterator.NextWithContext(ctx); err != nil {
+			return "", fmt.Errorf("Error listing Management Groups: %+v", err)
+		}
+	}
+
+	// we found none
+	if len(results) == 0 {
+		return "", fmt.Errorf("do not find Management Group (Display Name %q)", displayName)
```
[the remark this answers is not in the document]
Comment (ArcturusZhang) [reply azurerm/internal/services/managementgroup/management_group_data_source.go:148]: Done

File: azurerm/internal/services/managementgroup/management_group_data_source.go [hunk 7]
```diff
@@ -112,6 +125,37 @@ func dataSourceArmManagementGroupRead(d *schema.ResourceData, meta interface{})
...
+	}
+
+	// we found more than one
+	if len(results) > 1 {
+		return "", fmt.Errorf("found more than one Management Group with display name %q", displayName)
```
[the remark this answers is not in the document]
Comment (ArcturusZhang) [reply azurerm/internal/services/managementgroup/management_group_data_source.go:153]: Done

File: azurerm/internal/services/managementgroup/tests/management_group_data_source_test.go [hunk 8]
```diff
@@ -41,3 +59,19 @@ data "azurerm_management_group" "test" {
...
 }
+
+func testAccDataSourceArmManagementGroup_basicByDisplayName(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_management_group" "test" {
+  display_name = "acctestmg-%d"
```
[the remark this answers is not in the document]
Comment (ArcturusZhang) [reply azurerm/internal/services/managementgroup/tests/management_group_data_source_test.go:70]: Done

File: website/docs/r/management_group.html.markdown (hunk 1 above)
Comment (ArcturusZhang) [reply website/docs/r/management_group.html.markdown:50]: Done

File: website/docs/r/management_group.html.markdown (hunk 2 above)
Comment (ArcturusZhang) [reply website/docs/r/management_group.html.markdown:46]: Done

File: website/docs/d/management_group.html.markdown (hunk 3 above)
Comment (ArcturusZhang) [reply website/docs/d/management_group.html.markdown:41]: Done

File: azurerm/internal/services/managementgroup/tests/management_group_data_source_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (ArcturusZhang) [reply azurerm/internal/services/managementgroup/tests/management_group_data_source_test.go:40]: Given that my account does not have the permission for creating mgmt group (therefore I cannot run the acc tests of mgmt groups), I added an import step to another data source `azurerm_dedicated_host` and it gives me a lot of errors:
```
=== RUN   TestAccDataSourceAzureRMDedicatedHost_basic
=== PAUSE TestAccDataSourceAzureRMDedicatedHost_basic
=== CONT  TestAccDataSourceAzureRMDedicatedHost_basic
--- FAIL: TestAccDataSourceAzureRMDedicatedHost_basic (212.13s)
    testing.go:640: Step 1 error: 10 problems:
        
        - Provider produced invalid object: Provider "azurerm" planned an invalid value for data.azurerm_dedicated_host.test during refresh: unsupported attribute "license_type".
        
        This is a bug in the provider, which should be reported in the provider's own issue tracker.
        - Provider produced invalid object: Provider "azurerm" planned an invalid value for data.azurerm_dedicated_host.test during refresh: unsupported attribute "sku_name".
        
        This is a bug in the provider, which should be reported in the provider's own issue tracker.
        - Provider produced invalid object: Provider "azurerm" planned an invalid value for data.azurerm_dedicated_host.test during refresh: unsupported attribute "platform_fault_domain".

        
        This is a bug in the provider, which should be reported in the provider's own issue tracker.
        - Provider produced invalid object: Provider "azurerm" planned an invalid value for data.azurerm_dedicated_host.test during refresh: unsupported attribute "dedicated_host_group_id".
        
        This is a bug in the provider, which should be reported in the provider's own issue tracker.
        - Provider produced invalid object: Provider "azurerm" planned an invalid value for data.azurerm_dedicated_host.test during refresh: unsupported attribute "auto_replace_on_failure".
        
        This is a bug in the provider, which should be reported in the provider's own issue tracker.
        - Provider produced invalid object: Provider "azurerm" planned an invalid value for data.azurerm_dedicated_host.test during refresh: missing required attribute "dedicated_host_group_name".
        
        This is a bug in the provider, which should be reported in the provider's own issue tracker.
        - Provider produced invalid object: Provider "azurerm" planned an invalid value for data.azurerm_dedicated_host.test during refresh: missing required attribute "resource_group_name".
        
        This is a bug in the provider, which should be reported in the provider's own issue tracker.
        - Provider produced invalid object: Provider "azurerm" planned an invalid value for data.azurerm_dedicated_host.test during refresh: .timeouts: unsupported attribute "create".
        
        This is a bug in the provider, which should be reported in the provider's own issue tracker.
        - Provider produced invalid object: Provider "azurerm" planned an invalid value for data.azurerm_dedicated_host.test during refresh: .timeouts: unsupported attribute "delete".
        
        This is a bug in the provider, which should be reported in the provider's own issue tracker.
        - Provider produced invalid object: Provider "azurerm" planned an invalid value for data.azurerm_dedicated_host.test during refresh: .timeouts: unsupported attribute "update".
        
        This is a bug in the provider, which should be reported in the provider's own issue tracker.
FAIL
``` 
But after I removed the `importStep`, all these errors are gone and test passes.

**Comment by ArcturusZhang** [comment]
Hi @tombuildsstuff I have resolved the comments, please have a look.
As for the import step, I suppose data source test cannot add import steps (which will lead to errors). I did an experiment with the data source test of `dedicated host`.

File: website/docs/d/management_group.html.markdown (hunk 5 above)
[the remark this answers is not in the document]
Comment (tombuildsstuff) [reply website/docs/d/management_group.html.markdown]: yes - we should probably go through and remove those, I guess - we also need to work through and simplify some of them e.g. https://www.terraform.io/docs/providers/azurerm/r/key_vault_secret.html

File: azurerm/internal/services/managementgroup/tests/management_group_data_source_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (tombuildsstuff) [reply azurerm/internal/services/managementgroup/tests/management_group_data_source_test.go:40]: 🤦 yes, you're right - sorry

**Pull request merged**